apps/web/src/views/Home/components/UserBanner/HarvestCard.tsx

```diff
@@ -90,8 +91,29 @@ const HarvestCard = () => {
       }
     }
 
-    onHarvestAll(v3Farms.map((farm) => farm.sendTx.tokenId))
-  }, [farmsWithStakedBalance, onHarvestAll, fetchWithCatchTxError, gasPrice, toastSuccess, t])
+    const v3Farms = (
+      farmsWithStakedBalance.filter((value) => value && 'sendTx' in value) as {
+        sendTx: {
+          to: string
+          tokenId: string
+        }
+      }[]
+    ).map((farm) => farm.sendTx.tokenId)
+    if (v3Farms.length > 0) {
+      onHarvestStart?.()
```

Comment:
I see another `onHarvestStart` call at line [70](https://github.com/pancakeswap/pancake-frontend/pull/8044/files?diff=split&w=1#diff-f5f8c590921de2037c9b99cf48397f5da6b9674c2c5908cdff7686e73acacae7R70). Is it gonna get called twice?

Reply:
isn't the first `onHarvestStart` gonna get called anyway? Even for the case when user only has v3 farms. Can we remove the second call inside the if expression?

And it seems a bit weird if user doesn't have v3 farms, the `onHarvestEnd` won't get called while `onHarvestStart` does